File: backend/src/WasteToValue.Api/Modules/Collections/Entities/HandoverProof.cs
namespace WasteToValue.Api.Modules.Collections.Entities;

public sealed class HandoverProof
{
    public Guid Id { get; set; }
    public Guid PickupRequestId { get; set; }
    public Guid PickupEventId { get; set; }
    public string ProofType { get; set; } = string.Empty;
    public string? StorageKey { get; set; }
    public string? VerificationHash { get; set; }
    public Guid? VerifiedBy { get; set; }
    public DateTimeOffset? VerifiedAt { get; set; }
    public DateTimeOffset CreatedAt { get; set; }

    public PickupRequest? PickupRequest { get; set; }
    public PickupEvent? PickupEvent { get; set; }
}

File: mobile/lib/features/collections/models/handover_proof.dart
class HandoverProof {
  const HandoverProof({
    required this.id,
    required this.pickupRequestId,
    required this.pickupEventId,
    required this.proofType,
    this.verifiedBy,
    this.verifiedAt,
    required this.createdAt,
  });

  final String id;
  final String pickupRequestId;
  final String pickupEventId;
  final String proofType;
  final String? verifiedBy;
  final DateTime? verifiedAt;
  final DateTime createdAt;

  factory HandoverProof.fromJson(Map<String, dynamic> json) {
    return HandoverProof(
      id: json['id'] as String,
      pickupRequestId: json['pickupRequestId'] as String,
      pickupEventId: json['pickupEventId'] as String,
      proofType: json['proofType'] as String,
      verifiedBy: json['verifiedBy'] as String?,
      verifiedAt: json['verifiedAt'] != null ? DateTime.parse(json['verifiedAt'] as String) : null,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
